src/Microsoft.VisualStudio.Services.Agent/StreamString.cs

```diff
@@ -50,8 +50,12 @@ public StreamString(Stream ioStream)
         public async Task<string> ReadStringAsync(CancellationToken cancellationToken)
         {            
             Int32 len = await ReadInt32Async(cancellationToken);
-            cancellationToken.ThrowIfCancellationRequested();            
-            if (len <= 0 || len > MaxStringSize)
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I am not seeing value of this throw. all the methods used in this ReadStringAsync() is already take cancellation token, i would expect them to throw. normally, we do cancellationToken.ThrowIfCancellationRequested(); in a method if most of the sub-method it call doesn't support cancellation token.
Ex:
void CopyFiles(source, dest,  cancellationToken)
{
var files=getAllFiles(folder);
foreach(var file in files)
{
cancellationToken.ThrowIfCancelled();
copy(file, dest); // this doesn't take cancellationtoken.
}
}